src/k2/module/k23si/Module.cpp

```diff
@@ -268,6 +268,42 @@ dto::Key K23SIPartitionModule::_getContinuationToken(const IndexerIterator& it,
     };
 }
 
+// Makes the SKVRecord and applies the request's filter to it. If Status is not OK,
+// it should be returned in the response. Otherwise bool in tuple is whether the filter passed
+std::tuple<Status, bool> K23SIPartitionModule::_doQueryFilter(dto::K23SIQueryRequest& request, 
+                                                              dto::SKVRecord::Storage& storage) {
+    // We know the schema name exists because it is validated at the beginning of handleQuery
+    auto schemaIt = _schemas.find(request.key.schemaName);
+    auto versionIt = schemaIt->second.find(storage.schemaVersion);
+    if (versionIt == schemaIt->second.end()) {
+        return std::make_tuple(dto::K23SIStatus::OperationNotAllowed(
+            "Schema version of found record does not exist"), false);
+    }
+
+    dto::SKVRecord record(request.collectionName, versionIt->second);
+    record.storage = storage.share();
+    bool keep = false;
+    Status status = dto::K23SIStatus::OK("");
+
+    try {
+        keep = request.filterExpression.evaluate(record);
+    }
+    catch(dto::NoFieldFoundException&) {
```

Comment:
these two exceptions seem to be no-op. Why bother catching them?

Reply:
It needs to be caught so it doesn't propagate out. The keep = false is a no-op though and can be removed.